catroid/src/androidTestCreateatschool/java/org/catrobat/catroid/createatschool/uiespresso/ui/activity/RestrictedLoginTest.java

```diff
@@ -0,0 +1,232 @@
+
+	@Rule
+	public BaseActivityInstrumentationRule<CreateAtSchoolMainMenuActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(CreateAtSchoolMainMenuActivity.class, true, false);
+
+	private List<String> allSettings = new ArrayList<>(Arrays.asList(Constants.RESTRICTED_LOGIN_ACCEPTED, Constants.CREATE_AT_SCHOOL_USER));
+	private Map<String, Boolean> initialSettings = new HashMap<>();
+	private IdlingResource idlingResource;
+
+	@Before
+	public void setUp() {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		for (String setting : allSettings) {
+			initialSettings.put(setting, sharedPreferences.getBoolean(setting, false));
+		}
+
+		SharedPreferences.Editor editor = sharedPreferences.edit();
+		editor.putBoolean(Constants.RESTRICTED_LOGIN_ACCEPTED, false);
+		editor.putBoolean(Constants.CREATE_AT_SCHOOL_USER, true);
+		editor.apply();
+
+		Reflection.setPrivateField(Constants.class, "DEFAULT_ROOT", Environment.getExternalStorageDirectory()
+				.getAbsolutePath() + "/Pocket Code");
+
+		baseActivityTestRule.launchActivity(null);
+
+		idlingResource = baseActivityTestRule.getActivity().getIdlingResource();
+		Espresso.registerIdlingResources(idlingResource);
+
+		UiTestUtils.setFakeToken(baseActivityTestRule.getActivity(), false);
+	}
+
+	@Test
+	public void loginDialogsUiBehaviorTest() {
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.cancel))
+				.check(doesNotExist());
+		pressBack();
+
+		onView(withText(R.string.dialog_restricted_login_title))
+				.check(matches(isDisplayed()));
+
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.password_forgotten))
+				.check(doesNotExist());
+		onView(withText(R.string.cancel))
+				.check(doesNotExist());
+
+		pressBack();
+
+		onView(allOf(withText(R.string.login), withResourceName("alertTitle")))
+				.check(matches(isDisplayed()));
+	}
+
+	@Test
+	public void restrictedLoginDialogLinkTest() {
+		String link = Locale.getDefault().getLanguage().equals(Locale.GERMAN.getLanguage())
+				? Constants.RESTRICTED_LOGIN_LINK_GERMAN
+				: baseActivityTestRule.getActivity().getString(R.string.schoolreg_link);
+
+		Intents.init();
+		Matcher<Intent> expectedIntent = allOf(hasAction(Intent.ACTION_VIEW), hasData(link));
+		intending(expectedIntent).respondWith(new Instrumentation.ActivityResult(0, null));
+		Espresso.onView(ViewMatchers.withId(R.id.dialog_restricted_login_text))
+				.perform(ViewActions.openLinkWithText(link));
+		intended(expectedIntent);
+		Intents.release();
+	}
+
+	@Test
+	public void restrictedLoginDialogNotShownAfterConfirmationTest() {
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		relaunchActivity();
```

Comment:
as already stated in one of the other PRs im not a huge fan of activity switching and restarting in tests.

is there an easier way to check this? set whatever flag or shared preference that remembers that the dialog has been discarded, and then launch activity to just check that the dialog is not displayed on launch?

Reply:
Well, alternatively I could check whether the RESTRICTED_LOGIN_ACCEPTED preference has been set to true. I've changed that.